odl/test/tomo/geometry/geometry_test.py

```diff
@@ -549,6 +550,48 @@ def test_fanbeam_frommatrix():
                                                    det_rad, sing_mat)
 
 
+def test_fanbeam_flying_focal_spot():
+    """Test the flying focal spot in 2d fan beam geometry."""
+    full_angle = np.pi
+    apart = odl.uniform_partition(0, full_angle, 10)
+    dpart = odl.uniform_partition(0, 1, 10)
+    src_rad = 10
+    det_rad = 5
+    # Source positions with flying focal spot should correspond to
+    # source positions of 2 geometries with different starting positions
+    shift1 = np.array([2, -3])
+    shift2 = np.array([-2, 3])
+    init = np.array([1, 0], dtype=np.float32)
+    geom_ffs = odl.tomo.FanBeamGeometry(
+        apart, dpart,
+        src_rad, det_rad,
+        src_to_det_init=init,
+        src_shift_func=lambda angle: flying_focal_spot(
+            angle, apart=apart,
+            shifts=[shift1, shift2]))
+    # angles must be shifted to match discretization of apart
+    ang1 = -full_angle / 20
+    apart1 = odl.uniform_partition(ang1, full_angle + ang1, 5)
+    ang2 = full_angle / 20
+    apart2 = odl.uniform_partition(ang2, full_angle + ang2, 5)
+
+    init1 = init + np.array([0, shift1[1]]) / (src_rad + shift1[0])
+    init2 = init + np.array([0, shift2[1]]) / (src_rad + shift2[0])
+    # radius also changes when a shift is applied
+    src_rad1 = np.linalg.norm(np.array([src_rad, 0]) + shift1)
+    src_rad2 = np.linalg.norm(np.array([src_rad, 0]) + shift2)
+    geom1 = odl.tomo.FanBeamGeometry(apart1, dpart, src_rad1, det_rad,
+                                     src_to_det_init=init1)
+    geom2 = odl.tomo.FanBeamGeometry(apart2, dpart, src_rad2, det_rad,
+                                     src_to_det_init=init2)
+
+    sp1 = geom1.src_position(geom1.angles)
+    sp2 = geom2.src_position(geom2.angles)
+    sp = geom_ffs.src_position(geom_ffs.angles)
+    assert all_almost_equal(sp[0::2], sp1)
+    assert all_almost_equal(sp[1::2], sp2)
+
+
```

Comment:
Also add a test where this is actually used so we see that it interacts well with the backend.

Reply:
I have a separate jupyter notebook, where I have verified visually that ray transform works as expected. Should I include this without numerical verification of the result? 

As a next step I was thinking to planning detector shifts (by analogy) and then it would be possible to do some numerical verification by defining equivalent geometries with different source and detector shifts.

Reply:
I'd really appreciate a numerical test (even a rough one) for code this complex.